src/main/java/com/frcteam3255/components/SN_DoubleSolenoid.java

```diff
@@ -0,0 +1,117 @@
+package com.frcteam3255.components;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+
+public class SN_DoubleSolenoid extends DoubleSolenoid {
+
+	Value forwardValue;
+	Value reverseValue;
+
+	/**
+	 * Wrapper for the WPILib DoubleSolenoid
+	 * <p>
+	 * Adds custom methods like {@link #setDeployed()} and {@link #getDeployed()}
+	 *
+	 * @param a_moduleType
+	 *            CTRE Pneumatics Control Module or REV Pneumatics Hub
+	 * @param a_forwardChannel
+	 *            The forward channel of the solenoid on the Pneumatics Module
+	 * @param a_reverseChannel
+	 *            The reverse channel of the solenoid on the Pneumatics Module
+	 */
+	public SN_DoubleSolenoid(PneumaticsModuleType a_moduleType, int a_forwardChannel, int a_reverseChannel) {
+		super(a_moduleType, a_forwardChannel, a_reverseChannel);
+		forwardValue = Value.kForward;
+		reverseValue = Value.kReverse;
+	}
+
+	/**
+	 * Wrapper for the WPILib DoubleSolenoid
+	 * <p>
+	 * Adds custom methods like {@link #setDeployed()} and {@link #getDeployed()}
+	 *
+	 * @param a_module
+	 *            The CAN ID of the Pneumatics module to use. Leave blank to use
+	 *            default ID, 0
+	 * @param a_moduleType
+	 *            CTRE Pneumatics Control Module or REV Pneumatics Hub
+	 * @param a_forwardChannel
+	 *            The forward channel of the solenoid on the Pneumatics Module
+	 * @param a_reverseChannel
+	 *            The reverse channel of the solenoid on the Pneumatics Module
+	 */
+	public SN_DoubleSolenoid(int a_module, PneumaticsModuleType a_moduleType, int a_forwardChannel,
+			int a_reverseChannel) {
+		super(a_module, a_moduleType, a_forwardChannel, a_reverseChannel);
+		forwardValue = Value.kForward;
+		reverseValue = Value.kReverse;
+	}
+
+	private Value getValue() {
+		return this.get();
+	}
+
+	/**
+	 * @return true if the solenoid is deployed, false if the solenoid is retracted
+	 */
+	public boolean isDeployed() {
+		return getValue() == forwardValue;
+	}
+
+	/**
+	 * @return true if the solenoid is retracted, false if the solenoid is deployed
+	 */
+	public boolean isRetracted() {
+		return !isDeployed();
+	}
+
+	private void rawSetDeployed() {
+		this.set(forwardValue);
+	}
+
+	private void rawSetReverse() {
+		this.set(reverseValue);
+	}
+
+	/**
+	 * Deploys the solenoid
+	 */
+	public void setDeployed() {
+		if (!isDeployed()) {
+			rawSetDeployed();
+		}
+	}
+
+	/**
+	 * Retracts the solenoid
+	 */
+	public void setRetracted() {
+		if (isDeployed()) {
+			rawSetReverse();
+		}
+	}
```

Comment:
If it's already deployed, does it matter if you tell it to deploy again? Same for retract?

Reply:
I decided that because I had thought that there was still something physical happening on the solenoid when you deploy it, whether or not it was already deployed. I'll remove that logic, as well as the `rawSetDeployed()` and `rawSetReverse()` methods and directly call `this.set(value)` in `setDeployed()` and `setRetracted()`